Mono.Debugging/Mono.Debugging.Evaluation/ObjectValueAdaptor.cs

```diff
@@ -785,8 +785,7 @@ public virtual CompletionData GetExpressionCompletionData (EvaluationContext ctx
 					}
 
 					// FIXME: handle types and namespaces...
-				} catch (Exception ex) {
-					ctx.WriteDebuggerError (ex);
+				} catch {
```

Comment:
If the evaluation method throws due to a parsing or evaluation error (EvaluatorException), then it is OK to ignore that exception. But if it throws due to some internal error, that exception should be logged.

If the NRefactory exceptions mentioned in the bug report are parsing or evaluation errors then they should not leak through the Evaluate method. They should be cached and and a corresponding EvaluatorException should be thrown instead.